src/main/scala/io/pravega/connectors/spark/PravegaMicroBatchStream.scala

```diff
@@ -129,4 +130,30 @@ class PravegaMicroBatchStream(
     s"PravegaMicroBatchStream{clientConfig=${clientConfig}, scopeName=${scopeName}, streamName=${streamName}" +
       s" startStreamCut=${startStreamCut}, endStreamCut=${endStreamCut}}"
   }
+
+  override def latestOffset(start: Offset, readLimit: ReadLimit): Offset = {
+    val startOffset = Option(start)
+      .map(_.asInstanceOf[PravegaSourceOffset].streamCut).get
+    val limits: Seq[ReadLimit] = readLimit match {
+      case rows => Seq(rows)
+    }
+    val nextStreamCut = if (limits.exists(_.isInstanceOf[ReadAllAvailable])) {
+      PravegaSourceOffset(PravegaUtils.getStreamInfo(streamManager, scopeName, streamName).getTailStreamCut)
+    } else {
+      val upperLimit = limits.find(_.isInstanceOf[ReadMaxRows]).map(_.asInstanceOf[ReadMaxRows])
+      PravegaSourceOffset(batchClientFactory.getNextStreamCut(startOffset, upperLimit.get.maxRows()))
```

Comment:
1. How do we handle the exception here?
2. The rows maps to Pravega events, but now it's bytes being calculated in the call. Will that be changed?

Reply:
Will get back on this after discussion.

Reply:
1. Exception handlling will be done on pravega side.
 2. We are treating the maxoffset as bytes and returning next streamcut based on its value. This will not be changed and it will declare the same with this release.

Reply:
> We are treating the maxoffset as bytes.

This is not good as the API call is returning rows, this may introduce confusion for spark app developers. If it has to be the case, better to rename it as max bytes instead of offsets to emphasize the difference.
And according to the Pravega PR, it seems that this `max` definition is not fully guaranteed, in some cases it may consume more bytes, which is again confusing.

Reply:
Changed name to APPROX_BYTES_PER_TRIGGER. In documentation also we will mention this behavior.
If you have any better name, please provide suggestion.

Reply:
I think this is the best Pravega can do now here.

Reply:
According to the discussion here: https://github.com/pravega/pravega/pull/7230/files#r1318010466, we also need to handle the exception for this call.

Reply:
done